### PR TITLE
glass: show disk solution during installer

### PR DESCRIPTION
**Pull request by jecluis**
Instead of just showing the list of available devices, show instead the proposed solution.

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 1]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
-      next: (devices: Volume[]) => {
-        this.devices = devices;
+    this.nodesService.deploymentDiskSolution().subscribe({
+      next: (solution: DiskSolution) => {
+        const entries: TableEntry[] = [];
+        if (solution.systemdisk) {
+          const entry = this.consumeDisk(solution.systemdisk);
+          entry.isAvailable = true;
+          entry.isSystemDisk = true;
+          entry.useAs = 'system';
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:96]:
```suggestion
          entry.useAs = translate(TEXT('System'));
```

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 2]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
+          entries.push(entry);
+        }
+        solution.storage.forEach((disk: Disk) => {
+          const entry = this.consumeDisk(disk);
+          entry.isStorageDisk = true;
+          entry.isAvailable = true;
+          entry.useAs = 'storage';
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:103]:
```suggestion
          entry.useAs = translate(TEXT('Storage'));
```

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 3]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
+          entry.rejectedReasons = rejected.reasons;
+          entries.push(entry);
+        });
+        this.disks = entries;
       }
     });
   }
+
+  private consumeDisk(disk: Disk): TableEntry {
+    let typeStr = 'unknown';
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:118]:
```suggestion
    let typeStr = translate(TEXT('Unknown'));
```

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 4]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
+    if (disk.type === DiskTypeEnum.hdd) {
+      typeStr = 'HDD';
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:120]:
```suggestion
      typeStr = translate(TEXT('HDD'));
```

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 5]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
+    } else if (disk.type === DiskTypeEnum.ssd) {
+      typeStr = 'SSD';
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:122]:
```suggestion
      typeStr = translate(TEXT('SSD'));
```

File: src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts [hunk 6]
```diff
@@ -45,30 +61,75 @@ export class LocalDevicesStepComponent implements OnInit {
...
+    }
+    return {
+      path: disk.path,
+      size: disk.size.toString(),
+      type: typeStr,
+      useAs: 'N/A',
```
Comment (votdev) [inline src/glass/src/app/pages/install-wizard/steps/local-devices-step/local-devices-step.component.ts:128]:
```suggestion
      useAs: translate(TEXT('N/A')),
```

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by jecluis** [comment]
jenkins run tumbleweed

**Review by votdev — APPROVED** [review]

**Pull request merged**